### DISCUSSION TITLE
Disconnect user from Voice Channel

### DISCUSSION THREAD

**Discussion by TryhardOne** [Discord4J/Discord4J#925] [Q&A] [answered]
Hello, I was trying to add a function to my bot, so that if a user is self deaf in a channel for a certain time, my bot will kick him.

What I tried was:

	private void kickUser(Member m) {

		m.getVoiceState().block().getChannel().block().getVoiceConnection().block().disconnect().block();

	}

However, doing this results in an NullPointerException: reactor.core.publisher.Mono.block() is null
How would I do this?

**Comment by darichey** [comment, marked as answer]
Hi, you can use `GuildMemberEditSpec#setNewVoiceChannel(@Nullable Snowflake)`. For example, your method would be written as
```java
private void kickUser(Member m) {
  m.edit(spec -> spec.setNewVoiceChannel(null)).block();
}
```

Your attempt is actually trying to disconnect the bot from the channel. See `VoiceChannel#getVoiceConnection()`. The exception is probably because the bot is not in the channel.

**Reply by TryhardOne** [reply]
ah yes, that makes sense, works, thanks :3
